code/orphaned_procs/AStar.dm

```diff
@@ -123,23 +122,25 @@ Actual Adjacent procs :
 				continue
 
 			var/newg = cur.g + call(cur.source,dist)(T)
-			if(!T.PNode) //is not already in open list, so add it
-				open.Insert(new /PathNode(T,cur,newg,call(T,dist)(end),cur.nt+1))
+
+			var/PathNode/P = pnodelist[T]
```

Comment:
associated retrievals do not trigger the bad index runtime if it doesn't exist, it returns null.